ethereum/core/src/main/java/tech/pegasys/teku/core/blockvalidator/SimpleBlockValidator.java

```diff
@@ -120,13 +122,14 @@ public SimpleBlockValidator(
 
   private void verify_block_signature(final BeaconState state, SignedBeaconBlock signed_block)
       throws BlockProcessingException {
-    final Validator proposer =
-        state.getValidators().get(get_beacon_proposer_index(state, signed_block.getSlot()));
+    final int proposerIndex = get_beacon_proposer_index(state, signed_block.getSlot());
+    final BLSPublicKey proposerPublicKey =
+        ValidatorsUtil.getValidatorPubKey(state, UInt64.valueOf(proposerIndex)).orElseThrow();
```

Comment:
nit: Can we make this throw `BlockProcessingException` specifically.  It shouldn't ever happen but if it does it should be the same as the signature being invalid to ensure it gets handled correctly.
```suggestion
        ValidatorsUtil.getValidatorPubKey(state, UInt64.valueOf(proposerIndex)).orElseThrow(() -> new BlockProcessingException("Public key not found for validator " + proposerIndex));
```

Reply:
Let it be, thought that exception would still indicate internal logic malfunction cause `get_beacon_proposer_index` should normally return existing validator index